docs/src/explanations/connectable.md

```diff
@@ -0,0 +1,440 @@
+---
+
+layout: docs
+
+title:  "Connectable Operators"
+
+section: "chisel3"
+
+---
+
+## Terminology
+
+- "Chisel type" - a `Data` that is not bound to hardware
+  - E.g. `UInt(3.W)`, `new Bundle {..}`, `Vec(3, SInt(2.W))` are all Chisel types
+- `Aggregate` - a Chisel type or component that contains other Chisel types or components (i.e. Vec, Record, Bundle)
+- `Element` - a Chisel type or component that does not contain other Chisel types or components (e.g. UInt, SInt, Clock, Bool etc.)
+- "component" - a `Data` that is bound to hardware (`IO`, `Reg`, `Wire`, etc.)
+  - E.g. `Wire(UInt(3.W))` is a component, whose type is `UInt(3.W)`
+- "member" - a child Chisel type or component of a parent Chisel type or component (could be an `Aggregate` or an `Element`)
```

Comment:
so is `member` meaningless for Wire(Bool())? if so we should say it is part of an `Aggregate` parent chisel type?

Reply:
I think technically, a member can include the parent, so I updated the definition accordingly.